### DISCUSSION TITLE
Is there any way to trigger refresh of door status (e.g. door left open)

### DISCUSSION THREAD

**Discussion by dredjohn21** [G-Two/homeassistant-subaru#60] [Q&A]
I love this integration, but one thing I’ve noticed is the door and window status sometimes is not up to date (e.g. will show passenger door was left open but it’s not).

Is there any way to force a refresh to see what the current status is?  I could then trigger an alert to check the car at night if anything is left open?

**Comment by G-Two** [comment]
Yes, you can press the "Poll Vehicle" button manually. For automations, you can call the "Press Button" service. When actuated, this will request the Subaru telematics service to poll your vehicle and update all available data (including door open/close state). It generally takes about 5-10 seconds for the update to complete. The "Refresh" button is used to only request the Subaru API provide the latest cached data (i.e. it doesn't result in the vehicle being polled, and it may be stale data). Refresh functionality happens automatically every 5 minutes. Subarus will automatically send a data update when they are turned off (which is what the automatic 5 minute refresh is intended to capture). Polling will effectively force another vehicle update (which seems to be what your question is about). A nightly poll to check that your doors are closed should do what you need.

![image](https://user-images.githubusercontent.com/7310260/205509162-7f6aa329-4b35-492a-a15f-ae8a11ad0cd1.png)

This button was previously named "Locate" in versions prior to v0.7.0, and the (now deprecated) service was named "Update", which weren't great names.

**Reply by stboch** [reply]
I just trigger a lock every night, which will 1 lock the car if I forget and update the door and window status.

**Reply by IIAIronWolf** [reply]
> Yes, you can press the "Poll Vehicle" button manually. For automations, you can call the "Press Button" service. When actuated, this will request the Subaru telematics service to poll your vehicle and update all available data (including door open/close state). It generally takes about 5-10 seconds for the update to complete. The "Refresh" button is used to only request the Subaru API provide the latest cached data (i.e. it doesn't result in the vehicle being polled, and it may be stale data). Refresh functionality happens automatically every 5 minutes. Subarus will automatically send a data update when they are turned off (which is what the automatic 5 minute refresh is intended to capture). Polling will effectively force another vehicle update (which seems to be what your question is about). A nightly poll to check that your doors are closed should do what you need.
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/7310260/205509162-7f6aa329-4b35-492a-a15f-ae8a11ad0cd1.png)
> 
> 
> 
> This button was previously named "Locate" in versions prior to v0.7.0, and the (now deprecated) service was named "Update", which weren't great names.

Personally, this doesn't usually work for me. The API continues to report unlocked driver door or otherwise even after I poll it.

**Comment by G-Two** [comment]
@IIAIronWolf can you paste some of your API output (as described [here](https://github.com/G-Two/homeassistant-subaru/releases/tag/v0.7.0))? The Subaru API is inconsistent, and if they are using another value for certain cars (such as "CLOSED_LOCKED"), then we need to know. Right now the underlying [package](https://github.com/G-Two/subarulink/blob/218bf2dd8d4b2a18a7ea2593a305ac48382aa57e/subarulink/const.py#L52) is only aware of "OPEN" or "CLOSED". And the [integration](https://github.com/G-Two/homeassistant-subaru/blob/31321149e9cd578aa9745b87249a99e320d1fb28/custom_components/subaru/binary_sensor.py#L47) only looks for "OPEN" to determine the state of the binary sensor.

As of now, subarulink and this integration have never had the means of detecting unlocked/locked state. If you can provide an API dump that shows what values they are using, we can add door lock status, which would be great.
